include/dxc/Support/dxcapi.use.h

```diff
@@ -178,6 +193,64 @@ void WriteOperationErrorsToConsole(IDxcOperationResult *pResult,
 void WriteOperationResultToConsole(IDxcOperationResult *pRewriteResult,
                                    bool outputWarnings);
 
+class DxcDllExtValidationSupport : public DxcDllSupport {
+  // DxcDllExtValidationSupport manages the
+  // lifetime of dxcompiler.dll, while the member, m_DxilSupport,
+  // manages the lifetime of dxil.dll
+  DxcDllSupport *m_DxilSupport = nullptr;
+
+  std::string DxilDLLPathExt = "";
+  bool ExternalDxilDLLInitializationSuccess = false;
+  // override DxcDllSupport's implementation of InitializeInternal,
+  // adding the environment variable value check for a path to a dxil.dll
+  // for external validation
+  HRESULT InitializeInternal(LPCSTR dllName, LPCSTR fnName){
+
+    // Load dxcompiler.dll
+    HRESULT result = InitializeForDll(dllName, fnName);
+    // if dxcompiler.dll fails to load, then we won't try loading dxil.dll,
+    // so set this boolean to false.
+    ExternalDxilDLLInitializationSuccess = DXC_FAILED(result) ? false : true;
+    if (!ExternalDxilDLLInitializationSuccess) {
+      return result;
+    }
+
+    // now handle external dxil.dll
+    const char *envVal = std::getenv("DXC_DXIL_DLL_PATH");
+    bool ValidateInternally = false;
+    if (!envVal || std::string(envVal).empty()) {
+      ValidateInternally = true;
+    }
+
+    if (!ValidateInternally){
+      std::string DllPathStr(envVal);
+      DxilDLLPathExt = DllPathStr;
+      std::filesystem::path DllPath(DllPathStr);
+
+      // Check if path is absolute and exists
+      if (!DllPath.is_absolute() || !std::filesystem::exists(DllPath)) {
+        ExternalDxilDLLInitializationSuccess = false;
+        // TODO: Ideally emit some diagnostic that the given absolute path doesn't exist
+        return HRESULT_FROM_WIN32(GetLastError());
```

Comment:
This should simply result in a state where `!DxilDLLPathExt.empty() && !m_DxilSupport.IsEnabled()`, which we should have an accessor to detect as outlined.  We shouldn't return a failed state from this API, I don't think, since we did attach a dxcompiler.dll.  Code that uses this should check for this new failure condition after `Initialize()`.